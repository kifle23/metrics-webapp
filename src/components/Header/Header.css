header {
    background-color: #ec4c8a;
    color: #fff;
    padding: 0.7rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
    z-index: 3;
}

header h3 {
    font-weight: 100;
    font-size: 18px;
}

.icons {
    display: flex;
    gap: 1rem;
}

.previous {
    background: url(../../assets/icons/previous.png) no-repeat center center/cover;
    width: 10px;
    height: 18px;
    display: block;
    filter: invert(1);
}

.mic {
    background: url(../../assets/icons/microphone.png) no-repeat center center/cover;
    width: 20px;
    height: 20px;
    display: block;
    filter: invert(1);
}

.settings {
    background: url(../../assets/icons/setting.png) no-repeat center center/cover;
    width: 20px;
    height: 20px;
    display: block;
    filter: invert(1);
}

i {
    transition: all 0.2s ease-out;
}

i:active {
    filter: invert(0.2);
}

.settings:hover {
    transform: rotate(45deg);
}