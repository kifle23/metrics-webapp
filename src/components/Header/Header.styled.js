import styled from 'styled-components';

export const HeaderWrapper = styled.div`
`;
