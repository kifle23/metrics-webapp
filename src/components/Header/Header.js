import React from 'react';
import PropTypes from 'prop-types';
import { HeaderWrapper } from './Header.styled';

const Header = () => (
 <HeaderWrapper>
    Header Component
 </HeaderWrapper>
);

Header.propTypes = {};

Header.defaultProps = {};

export default Header;
