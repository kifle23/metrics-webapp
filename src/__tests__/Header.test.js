import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Header from '../components/Header/Header';

describe('Header component', () => {
  test('renders without errors', () => {
    const component = renderer.create(
      <Router>
        <Header heading="test" previous="" />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the heading element', () => {
    render(
      <Router>
        <Header heading="test" previous="" />
      </Router>,
    );
    const headingElement = screen.getByText('test');
    expect(headingElement).toBeInTheDocument();
  });
});
