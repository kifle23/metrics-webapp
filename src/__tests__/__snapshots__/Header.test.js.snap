// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create header Render without errors 1`] = `
<header>
  <h3>
    test
  </h3>
  <div
    className="icons"
  >
    <i
      className="mic"
    />
    <i
      className="settings"
    />
  </div>
</header>
`;

exports[`Header component renders without errors 1`] = `
<header>
  <h3>
    test
  </h3>
  <div
    className="icons"
  >
    <i
      className="mic"
    />
    <i
      className="settings"
    />
  </div>
</header>
`;
